app/common/renderer/actions/Session.js

```diff
@@ -1190,3 +1215,35 @@ export function initFromQueryString(loadNewSession) {
     }
   };
 }
+
+export function setEnvironmentVariables(envVars) {
+  return async (dispatch) => {
+    await setSetting(ENVIRONMENT_VARIABLES, envVars);
+    dispatch({type: SET_ENVIRONMENT_VARIABLES, envVars});
+  };
+}
+
+export function loadEnvironmentVariables() {
+  return async (dispatch) => {
+    const envVars = (await getSetting(ENVIRONMENT_VARIABLES)) || [];
+    dispatch({type: SET_ENVIRONMENT_VARIABLES, envVars});
+  };
+}
+
+export function addEnvironmentVariable(key, value) {
+  return async (dispatch, getState) => {
+    const currentEnvVars = getState().inspector.environmentVariables || [];
```

Comment:
I don't see why these are retrieved from the inspector reducer (they aren't relevant anymore at that point)
```suggestion
    const currentEnvVars = getState().session.environmentVariables || [];
```